packages/js-dpp/lib/identity/validation/validatePublicKeysFactory.js

```diff
@@ -97,11 +98,33 @@ function validatePublicKeysFactory(validator) {
     // validate key data
     rawPublicKeys
       .forEach((rawPublicKey) => {
-        const dataHex = rawPublicKey.data.toString('hex');
-
-        if (!PublicKey.isValid(dataHex)) {
-          const validationError = PublicKey.getValidationError(dataHex);
+        let validationError;
+
+        switch (rawPublicKey.type) {
+          case IdentityPublicKey.TYPES.ECDSA_SECP256K1: {
+            const dataHex = rawPublicKey.data.toString('hex');
+
+            if (!PublicKey.isValid(dataHex)) {
+              validationError = PublicKey.getValidationError(dataHex);
+            }
+            break;
+          }
+          case IdentityPublicKey.TYPES.BLS12_381: {
+            try {
+              bls.PublicKey.fromBytes(
+                Uint8Array.from(rawPublicKey.data),
+              );
+            } catch (e) {
+              validationError = new TypeError('Invalid public key');
+            }
+            break;
+          }
+          case IdentityPublicKey.TYPES.ECDSA_HASH160:
+          default:
+            // Do nothing
```

Comment:
I would throw an error in case unknown type in case if we forgot to implement new type here again ))